### ISSUE TITLE
The implementation seems incorrect?

### ISSUE THREAD

**Issue by hzhwcmhf** [neural-dialogue-metrics/Distinct-N#2]
As far as I know, Distinct should be calculated in corpus-level and should not average over the sentences.
That is, we count the n-gram over the corpus, and then calculate the rate.

In this implmentation, ``Distinct`` will be higher than the value reported in Li's paper.
Maybe there is a better reference and they report the number of n-gram. https://arxiv.org/pdf/1606.08340.pdf

**Comment by Shijie97** [comment]
> As far as I know, Distinct should be calculated in corpus-level and should not average over the sentences.
> That is, we count the n-gram over the corpus, and then calculate the rate.
> 
> In this implmentation, `Distinct` will be higher than the value reported in Li's paper.
> Maybe there is a better reference and they report the number of n-gram. https://arxiv.org/pdf/1606.08340.pdf

Hello thx for your constructive comments and I'd love to ask u if u know where there were codes that implemented distinct-1/1 algorithm.

**Comment by iseesaw** [comment]
Yeah, the value will be very high if calculated per utterance. [Here](https://github.com/PaddlePaddle/models/blob/release/1.6/PaddleNLP/Research/Dialogue-PLATO/plato/metrics/metrics.py) is an implementation by BaiduNLP.
